src/components/app/details/cdDetails/service.ts

```diff
@@ -138,11 +138,18 @@ export const prepareConfigMapAndSecretData = (
             secretValues['external'] = { displayName: 'Data type', value: EXTERNAL_TYPES[type][''] }
             if (type === 'Secret' && historyData.codeEditorValue.value) {
                 const secretData = JSON.parse(historyData.codeEditorValue.value)
-                const decodeNotRequired =
+                const resolvedSecretData = JSON.parse(historyData.codeEditorValue.resolvedValue)
+                let decodeNotRequired =
                     skipDecode || Object.keys(secretData).some((data) => secretData[data] === '*****') // Don't decode in case of non admin user
                 historyData.codeEditorValue.value = decodeNotRequired
                     ? historyData.codeEditorValue.value
                     : JSON.stringify(decode(secretData))
+
+                decodeNotRequired =
```

Comment:
We can use the above value. No need to recalculate the same

Reply:
resolved